allennlp/models/coreference_resolution/coref.py

```diff
@@ -400,8 +401,9 @@ def _compute_antecedent_scores(self,
         antecedent_scores += antecedent_log_mask
 
         # Shape: (batch_size, num_spans_to_keep, 1)
-        dummy_scores = util.get_zeros([antecedent_scores.size(0), antecedent_scores.size(1), 1],
-                                      antecedent_scores.is_cuda)
+        shape = [antecedent_scores.size(0), antecedent_scores.size(1), 1]
+        dummy_scores = Variable(antecedent_scores.data.new().resize_(shape).fill_(0),
```

Comment:
I think we should try to avoid having to pass the device to functions where possible - I wanted to make this work for the `get_range_vector` as well, but it was used in more places and seemed more necessary to have the device passed to the function. Maybe I should change it back, unsure.

Reply:
When I wrote that, I was thinking the function did something like our `ones_like` function.  I agree with not liking passing around `is_cuda` flags.

Reply:
If you think it'd be cleaner to have a `zeros_like`, even with an optional `shape` parameter, or something, that's fine with me.

Reply:
Or even `constant_tensor_like`, with `shape` and `value`  as parameters...  Not sure how far to take this.